include/zephyr/kernel.h

```diff
@@ -545,9 +545,11 @@ __syscall int k_thread_join(struct k_thread *thread, k_timeout_t timeout);
  *
  * @param timeout Desired duration of sleep.
  *
- * @return Zero if the requested time has elapsed or if the thread was woken up
- * by the \ref k_wakeup call, the time left to sleep rounded up to the nearest
- * millisecond.
+ * @retval 0 Requested time has elapsed
+ * @retval >0 Number of milliseconds (rounded up) left to sleep if prematurely awakened
+ * @retval -ERANGE Remaining time to sleep is too far into the future to represent
```

Comment:
This really seems odd, if -ERANGE is possible making k_sleep fallible now (which previously it wasn't?) then something would seemingly be *wrong* with the way we are passing a timeout. It's fairly non-obvious what values might result in -ERANGE given k_timeout_t is a type alias, with macros and such, that we ourselves are defining.

There's a *lot* of calls to k_sleep without checking returns in the wild I'd expect. Are we expecting everyone to now add assert/branching around this to check?

Reply:
Yeah, it's an API bug at its core.  `k_sleep()` takes a k_timeout_t argument which is a potentially 64 bit count of ticks, but returns an integer number of **milliseconds** for historical reasons (because it was originally specified in ms on both ends).   There was an easy migration to timeout values for the argument via the various `K_<time_unit>()` macros, but none of that works to fix code that already exists expecting a number in a specific unit.